### ISSUE TITLE
Build script that checks for i128 support doesn't seem to work for Web Assembly

### ISSUE THREAD

**Issue by Emiluren** [servo/bincode#292]
I tried to compile a project that uses bincode to Web Assembly (`--target=wasm32-unknown-emscripten`) but it does not compile. I get several error messages related to i128 and u128:

```
error[E0407]: method `deserialize_u128` is not a member of trait `serde::Deserializer`
   --> /home/emil/.cargo/registry/src/github.com-1ecc6299db9ec823/bincode-1.2.0/src/de/mod.rs:65:9
    |
65  | /         fn $dser_method<V>(self, visitor: V) -> Result<V::Value>
66  | |             where V: serde::de::Visitor<'de>,
67  | |         {
68  | |             try!(self.read_type::<$ty>());
69  | |             let value = try!(self.reader.$reader_method::<O::Endian>());
70  | |             visitor.$visitor_method(value)
71  | |         }
    | |_________^ not a member of trait `serde::Deserializer`
...
112 |       impl_nums!(u128, deserialize_u128, visit_u128, read_u128);
    |       ---------------------------------------------------------- in this macro invocation

error[E0407]: method `deserialize_i128` is not a member of trait `serde::Deserializer`
   --> /home/emil/.cargo/registry/src/github.com-1ecc6299db9ec823/bincode-1.2.0/src/de/mod.rs:65:9
    |
65  | /         fn $dser_method<V>(self, visitor: V) -> Result<V::Value>
66  | |             where V: serde::de::Visitor<'de>,
67  | |         {
68  | |             try!(self.read_type::<$ty>());
69  | |             let value = try!(self.reader.$reader_method::<O::Endian>());
70  | |             visitor.$visitor_method(value)
71  | |         }
    | |_________^ not a member of trait `serde::Deserializer`
...
115 |       impl_nums!(i128, deserialize_i128, visit_i128, read_i128);
    |       ---------------------------------------------------------- in this macro invocation

error[E0407]: method `serialize_u128` is not a member of trait `serde::Serializer`
  --> /home/emil/.cargo/registry/src/github.com-1ecc6299db9ec823/bincode-1.2.0/src/ser/mod.rs:92:5
   |
92 | /     fn serialize_u128(self, v: u128) -> Result<()> {
93 | |         self.writer.write_u128::<O::Endian>(v).map_err(Into::into)
94 | |     }
   | |_____^ not a member of trait `serde::Serializer`

error[E0407]: method `serialize_i128` is not a member of trait `serde::Serializer`
  --> /home/emil/.cargo/registry/src/github.com-1ecc6299db9ec823/bincode-1.2.0/src/ser/mod.rs:97:5
   |
97 | /     fn serialize_i128(self, v: i128) -> Result<()> {
98 | |         self.writer.write_i128::<O::Endian>(v).map_err(Into::into)
99 | |     }
   | |_____^ not a member of trait `serde::Serializer`

error[E0599]: no method named `visit_u128` found for type `V` in the current scope
   --> /home/emil/.cargo/registry/src/github.com-1ecc6299db9ec823/bincode-1.2.0/src/de/mod.rs:112:40
    |
112 |     impl_nums!(u128, deserialize_u128, visit_u128, read_u128);
    |                                        ^^^^^^^^^^ help: there is a method with a similar name: `visit_u16`

error[E0599]: no method named `visit_i128` found for type `V` in the current scope
   --> /home/emil/.cargo/registry/src/github.com-1ecc6299db9ec823/bincode-1.2.0/src/de/mod.rs:115:40
    |
115 |     impl_nums!(i128, deserialize_i128, visit_i128, read_i128);
    |                                        ^^^^^^^^^^ help: there is a method with a similar name: `visit_i16`
```

It was my understanding that [this line](https://github.com/servo/bincode/blob/12f1415b9ec5d2eec7712784da7690571b56a7d6/build.rs#L7) was supposed to automatically check for i128 support

**Comment by jdm** [comment]
Might be related to #285.

**Comment by tkaitchuck** [comment]
@jdm I don't think those are related. Though this likely is: https://github.com/servo/bincode/blob/b676754eeee41d169e96bcaf211ed49a331b0d87/Cargo.toml#L27

**Comment by Emiluren** [comment]
This seems to have been solved by bincode 1.2.1 / #281
